### PR TITLE
Create docker image for updating the cache of Serlo's API

### PR DESCRIPTION
**Pull request by hugotiburtino**
Continuation of https://github.com/serlo/api.serlo.org/pull/75

### CODE REVIEW

**Comment by hugotiburtino** [comment]
I would like to close one chapter of this story.
In other PRs, I would like to do the CICD as well add the feature of bisect errors, and improve the design/performance using queue.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  extends: [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    "prettier/@typescript-eslint",
+    "prettier",
+  ],
+  parser: "@typescript-eslint/parser",
+  parserOptions: {
+    project: ["tsconfig.json"],
```
Comment (inyono) [inline .eslintrc.js]: Just FYI if you run into this problem later: when you have any JS files in the root of your project (e.g. config files), you will need `tsconfig.eslint.json` here (as in api.serlo.org).

File: src/index.ts [hunk 2]
```diff
@@ -19,56 +19,51 @@
...
+function start() {
   const cacheWorker = new CacheWorker({
     apiEndpoint: process.env.SERLO_ORG_HOST as string,
-    secret: process.env.SERLO_ORG_SECRET,
-    service: process.env.SERLO_SERVICE as Service,
-    pagination,
-  })
+    secret: process.env.SECRET,
+    service: process.env.SERVICE,
+    pagination: parseInt(process.env.PAGINATION as string),
```
Comment (inyono) [inline src/index.ts]: You can save yourself some headache by adding an `env.d.ts` file that describes all available env variables and their type, see e.g. https://github.com/serlo/api.serlo.org/blob/master/src/env.d.ts. This way,
- you don't need to work with `undefined` (and therefore don't need to typecast)
- have documented which env variables exist (so we easily know what to pass to the Docker image)

File: tsconfig.json [hunk 3]
```diff
@@ -4,8 +4,8 @@
     "esModuleInterop": true,
     "importHelpers": true,
     "lib": ["esnext"],
-    "module": "commonjs",
```
Comment (inyono) [inline tsconfig.json:7]: Why is this change needed?

**Review by inyono — COMMENTED** [review]
Looks good. I created a follow-up issue for the prettier config (#2). Furthermore: I suppose we don't need the `.gitmodules` file anymore?

File: tsconfig.json (hunk 3 above)
Comment (hugotiburtino) [reply tsconfig.json:7]: Because it was conflicting with esnext. But I've spotted the problem and, like at api.serlo.org, I'm going to upload and use a tsconfig.prod.json.

Comment (inyono) [reply tsconfig.json:7]: Ah, yeah, that makes sense. Yes, please use the same build toolchain as in api.serlo.org since that worked out well for us in other projects, too :)

**Review by inyono — APPROVED** [review]
Thanks 👍
